lib/Model/Settings/AppSettings.php

```diff
@@ -132,6 +132,65 @@ public function getComboAllowed(): bool {
 		return false;
 	}
 
+	/**
+	 * Permission to create shares is controlled by core settings
+	 */
+	public function getShareCreateAllowed(): bool {
+		if (!$this->userSession->getIsLoggedIn()) {
+			// only logged in users can create shares
+			return false;
+		}
+
+		// first check group exception mode
+		$groupExceptionMode = $this->getShareGroupExceptionMode();
+
+		if ($groupExceptionMode === 'off') {
```

Comment:
While testing with `shareapi_exclude_groups_list` containing values, the function returns early here, without checking for excluded groups and allows users in the group to find other users  and create shares

Reply:
This is what the frontend tells me. 

If `shareapi_exclude_groups === 'no'` the list should get ignored. Of course just if the mode is requested correctly. 

![image](https://github.com/user-attachments/assets/43a6fd96-a90d-4464-a4ff-ef7ce2d34d06)